src/expr.rs

```diff
@@ -306,49 +307,81 @@ impl PyExpr {
     /// Extracts the Expr value into a PyObject that can be shared with Python
     pub fn python_value(&self, py: Python) -> PyResult<PyObject> {
         match &self.expr {
-            Expr::Literal(scalar_value) => Ok(match scalar_value {
```

Comment:
The main change here was to remove the `Ok` wrapping the result of the `match` and moving into the match arms so that we can have some of them return `Err` instead.